src/v/security/oidc_authenticator.cc

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "security/oidc_authenticator.h"
+
+#include "bytes/bytes.h"
+#include "config/property.h"
+#include "http/client.h"
+#include "json/document.h"
+#include "net/transport.h"
+#include "security/acl.h"
+#include "security/errc.h"
+#include "security/logger.h"
+#include "vlog.h"
+
+#include <seastar/http/client.hh>
+
+#include <boost/algorithm/string/split.hpp>
+#include <boost/outcome/success_failure.hpp>
+#include <boost/url/parse.hpp>
+#include <cryptopp/base64.h>
+#include <cryptopp/integer.h>
+#include <cryptopp/rsa.h>
+#include <fmt/chrono.h>
+#include <fmt/core.h>
+#include <fmt/ranges.h>
+
+#include <string_view>
+
+namespace security {
+
+std::ostream& operator<<(std::ostream& os, oidc_authenticator::state const s) {
+    using state = oidc_authenticator::state;
+    switch (s) {
+    case state::init:
```

Comment:
micro-nit: could `using enum`

Reply:
> micro-nit: could `using enum`

Oh, I moved that code from the impl. It's not needed.

Reply:
I prefer to keep the scope of scoped enums.